File: app/src/main/java/com/cbr/behancesampleapp/domain/network/BehanceRepository.java
package com.cbr.behancesampleapp.domain.network;

import com.cbr.behancesampleapp.BuildConfig;
import com.cbr.behancesampleapp.domain.model.BehanceSinlgeUserReponse;
import com.cbr.behancesampleapp.domain.model.BehanceUserResponse;

import java.util.Map;

import io.reactivex.Observable;

/** Created by dimitrios on 25/08/2017.*/
public class BehanceRepository extends DataRepository {

	private BehanceApiService mBehanceApiService;

	public BehanceRepository(BehanceApiService service) {
		this.mBehanceApiService = service;
	}

	public Observable<BehanceUserResponse> getUsers(Map<String, Object> params) {
		return mBehanceApiService.getUsers(params).compose(this.<BehanceUserResponse>applySchedulers());
	}

	public Observable<BehanceSinlgeUserReponse> getUserById(String userId) {
		return this.mBehanceApiService.getUserById(userId, BuildConfig.API_KEY).compose(this.<BehanceSinlgeUserReponse>applySchedulers());
	}

}

File: app/src/main/java/com/cbr/behancesampleapp/domain/network/BehanceRepository.kt
package com.cbr.behancesampleapp.domain.network

import com.cbr.behancesampleapp.BuildConfig
import com.cbr.behancesampleapp.domain.model.BehanceSingleUserReponse
import com.cbr.behancesampleapp.domain.model.BehanceUserResponse

import io.reactivex.Observable

/** Created by dimitrios on 25/08/2017. */
class BehanceRepository(private val mBehanceApiService: BehanceApiService) : DataRepository() {
    
    fun getUsers(params: MutableMap<String, Any>): Observable<BehanceUserResponse> {
        return mBehanceApiService.getUsers(params).compose(this.applySchedulers())
    }
    
    fun getUserById(userId: String): Observable<BehanceSingleUserReponse> {
        return this.mBehanceApiService.getUserById(userId, BuildConfig.API_KEY).compose(this.applySchedulers())
    }
    
}